include/engine/api/route_api.hpp

```diff
@@ -41,29 +41,35 @@ class RouteAPI : public BaseAPI
     {
     }
 
-    void MakeResponse(const InternalRouteResult &raw_route, util::json::Object &response) const
+    void MakeResponse(const InternalManyRoutesResult &raw_routes,
+                      util::json::Object &response) const
     {
-        auto number_of_routes = raw_route.has_alternative() ? 2UL : 1UL;
-        util::json::Array routes;
-        routes.values.resize(number_of_routes);
-        routes.values[0] = MakeRoute(raw_route.segment_end_coordinates,
-                                     raw_route.unpacked_path_segments,
-                                     raw_route.source_traversed_in_reverse,
-                                     raw_route.target_traversed_in_reverse);
-        if (raw_route.has_alternative())
+        BOOST_ASSERT(!raw_routes.routes.empty());
+
+        util::json::Array jsRoutes;
+
+        for (const auto &route : raw_routes.routes)
         {
-            std::vector<std::vector<PathData>> wrapped_leg(1);
-            wrapped_leg.front() = std::move(raw_route.unpacked_alternative);
-            routes.values[1] = MakeRoute(raw_route.segment_end_coordinates,
-                                         wrapped_leg,
-                                         raw_route.alt_source_traversed_in_reverse,
-                                         raw_route.alt_target_traversed_in_reverse);
+            if (!route.is_valid())
+                continue;
+
+            jsRoutes.values.push_back(MakeRoute(route.segment_end_coordinates,
+                                                route.unpacked_path_segments,
+                                                route.source_traversed_in_reverse,
+                                                route.target_traversed_in_reverse));
         }
-        response.values["waypoints"] = BaseAPI::MakeWaypoints(raw_route.segment_end_coordinates);
-        response.values["routes"] = std::move(routes);
+
+        response.values["waypoints"] =
+            BaseAPI::MakeWaypoints(raw_routes.routes[0].segment_end_coordinates);
+        response.values["routes"] = std::move(jsRoutes);
         response.values["code"] = "Ok";
     }
 
+    void MakeResponse(const InternalRouteResult &raw_route, util::json::Object &response) const
+    {
+        return MakeResponse(raw_route, response);
```

Comment:
Hm the InternalRouteResult should just silently convert to a InternalManyRouteResult, at least that was the plan. Not sure you're right because then basically no test would have run through. I will make a pr to change it to your suggestion - it's much clearer. Thanks for flagging!